public/javascripts/SVLabel/src/SVLabel/modal/ModalMission.js

```diff
@@ -110,7 +110,8 @@ function ModalMission (missionContainer, neighborhoodContainer, uiModalMission,
                 distanceString;
                 templateHTML = distanceMissionHTML;
 
-            if(missionContainer.isTheFirstMission()){
+            if ((missionContainer.onlyMissionOnboardingDone() || missionContainer.isTheFirstMission())
+                && !svl.storage.get('completedMissionAnonymously')) {
```

Comment:
Instead of checking it explicitly here using `!svl.storage.get('completedMissionAnonymously')`, why don't you add it to the `isTheFirstMission()`? Because logically it should be a part of the check done for first mission.